dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ClusterUtils.java

```diff
@@ -81,8 +82,14 @@ public static URL mergeUrl(URL remoteUrl, Map<String, String> localMap) {
             // So, generally, we don't need to care about the group value here.
             // But when comes to group merger, there is an exception, the consumer group may be '*' while the provider group can be empty or any other values.
             String remoteGroup = map.get(Constants.GROUP_KEY);
+            String remoteRelease = map.get(Constants.RELEASE_KEY);
             map.putAll(localMap);
             map.put(Constants.GROUP_KEY, remoteGroup);
+            // we should always keep the Provider RELEASE_KEY not overrode by the the value on Consumer side.
+            map.remove(Constants.RELEASE_KEY);

```

Comment:
If you use 
```
if (StringUtils.isNotEmpty(remoteRelease)) {
                map.put(Constants.RELEASE_KEY, remoteRelease);
            }else {
           map.remove(Constants.RELEASE_KEY);
}
```
It will be better ?